### PR TITLE
Migrate CI to GitHub Actions

### LINKED ISSUE #415: Move to Github Actions
Travis is very very slow. It would be nice to move to github actions.

### PR DESCRIPTION
**Pull request by taiki-e**
Closes #415

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    name: Publish Documentation
+    needs: [style, test]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Install Rust
+        run: rustup update beta && rustup default beta
+      - name: Build documentation
+        run: cargo doc --no-deps --all-features
+      - name: Publish documentation
```
Comment (taiki-e) [inline .github/workflows/CI.yml]: This is based on [the script used in stdarch.](https://github.com/rust-lang/stdarch/blob/master/.github/workflows/main.yml#L32-L39)

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  test:
+    name: Test
+    needs: [style]
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: -Dwarnings
+    strategy:
+      matrix:
+        rust:
+          - nightly
```
Comment (taiki-e) [inline .github/workflows/CI.yml]: GitHub Actions don't have allow_failures, so we probably need to use a way like https://stackoverflow.com/questions/57850553/github-actions-check-steps-status.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  style:
+    name: Check Style
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Install Rust
```
Comment (LucioFranco) [inline .github/workflows/CI.yml]: what about using a rust-toolchain file?

**Comment by LucioFranco** [comment]
@carllerche @seanmonstar Are you all +1 on this?

File: .github/workflows/CI.yml (hunk 3 above)
Comment (taiki-e) [reply .github/workflows/CI.yml]: H2 has some checks that require nightly...
https://github.com/hyperium/h2/blob/3cfcab016e17f36884ee77ff85947402535468ee/.travis.yml#L47-L51

**Pull request merged**